src/runtime_src/core/edge/skd/main.cpp

```diff
@@ -34,65 +38,85 @@
  * and exit.
  */
 
+using severity_level = xrt_core::message::severity_level;
+
 int main(int argc, char *argv[])
 {
-  pid_t pid, sid;
-  xclDeviceHandle handle;
-  xclSKCmd cmd;
-
-  pid = fork();
-  if (pid < 0) {
+  pid_t pid = fork();
+  if (pid < 0)
     exit(EXIT_FAILURE);
-  }
+
   if (pid > 0)
     exit(EXIT_SUCCESS);
 
+  // Starting the child process
   umask(0);
 
-  /* Open syslog and send the first message */
-  openlog("skd", LOG_PID | LOG_CONS, LOG_LOCAL0);
-  syslog(LOG_INFO, "Daemon Start...\n");
+  // Send the first message to XRT log
+  const std::string msg = "Daemon Start...";
+  xrt_core::message::send(severity_level::info, "SKD", msg);
 
-  /* Create a new SID for the child process */
-  sid = setsid();
+  // Create a new SID for the child process
+  pid_t sid = setsid();
   if (sid < 0) {
-    syslog(LOG_INFO, "Set SID failed. Daemon exiting\n");
-    closelog();
+    const auto errMsg = "Set SID failed. Daemon exiting";
+    xrt_core::message::send(severity_level::error, "SKD", errMsg);
     exit(EXIT_FAILURE);
   }
-  syslog(LOG_INFO, "SID set %d\n", sid);
+  const auto sid_msg = boost::format("SID set %d") % sid;
+  xrt_core::message::send(severity_level::info, "SKD", sid_msg.str());
 
   if (chdir("/") < 0) {
-    syslog(LOG_INFO, "Could NOT change to \"/\" directory\n");
-    closelog();
+    const auto errMsg = "Could NOT change to \"/\" directory";
+    xrt_core::message::send(severity_level::error, "SKD", errMsg);
     exit(EXIT_FAILURE);
   }
 
-  handle = initXRTHandle(0);
+  xclDeviceHandle handle = initXRTHandle(0);
   if (!handle) {
-    syslog(LOG_INFO, "Fail to init XRT\n");
-    closelog();
+    const auto errMsg = "Fail to init XRT";
+    xrt_core::message::send(severity_level::error, "SKD", errMsg);
     exit(EXIT_FAILURE);
   }
 
-  while (1) {
-    /* Calling XRT interface to wait for commands */
+  uint64_t mem_start_paddr = 0;
+  uint64_t mem_size = 0;
+  int parent_mem_bo = 0;
+#ifdef SKD_MAP_BIG_BO
+  // Map entire PS reserve memory space
+  xrtDeviceHandle xrtdHdl = xrtDeviceOpenFromXcl(handle);
+  try {
+    mem_size = xrt_core::device_query<xrt_core::query::host_mem_size>(xrt_core::device_int::get_core_device(xrtdHdl));
+    mem_start_paddr = xrt_core::device_query<xrt_core::query::host_mem_addr>(xrt_core::device_int::get_core_device(xrtdHdl));
+    parent_mem_bo = xclGetHostBO(handle,mem_start_paddr,mem_size);
+    syslog(LOG_INFO, "host_mem_size=%ld, host_mem_address=%lx\n",mem_size, mem_start_paddr);
+  }
+  catch (const xrt_core::query::exception& ex) {
+    xrt_core::send_exception_message(ex.what());
+  }
```

Comment:
Don't we need to return here propagating the error code to our caller?

Reply:
This is the main skd daemon, so there is nothing to return.